### PR TITLE
[RFC] Enable -Wconversion on quickfix.c

### PR DESCRIPTION
**Pull request by vheon**

### CODE REVIEW

**Comment by Pyrohh** [comment]
Hi @vheon,

Could you drop the `bool` changes, or perhaps move them to a separate PR? I've been seeing this recently and I consider it an anti-pattern: they have no relation to `-Wconversion`, and, in this case, nearly triple the size of the PR.

**Comment by vheon** [comment]
I thought that enabling the `-Wconversion` was an opportunities to use the right types for that TU and I made two separate commits exactly for remark that they are distinct set of changes. But if you don't want the `bool` refactor I could easily drop them; you have just to decide if you care more about the git log or the issue tracker.

**Comment by Pyrohh** [comment]
> I thought that enabling the -Wconversion was an opportunities to use the right types for that TU and I made two separate commits exactly for remark that they are distinct set of changes

From what I've heard in the past, enabling `-Wconversion` (along with tests) can be seen as a precursor to refactoring, a "cleaning" if you will. As far as I'm aware, using `bool` is mostly a stylistic choice (`stdbool.h` usually defines `_Bool` as an enum), i.e., it arguably won't help refactoring much. Just to be clear, this is solely my view on it, so maybe someone else should chime in (@justinmk @fmoralesc?).

> you have just to decide if you care more about the git log or the issue tracker.

I think there might be a misunderstanding (or I'm misreading you); I'm only suggesting it because it's easier to review PRs. I'd love it if we got rid of `TRUE` and `FALSE`, but IMO they're not an important enough concern to integrate them into `-Wconversion` PRs.

**Comment by Pyrohh** [comment]
Also, I'm aware that GCC 5.1 has a few new warnings regarding `bool`s, so I guess the choice to use them over `TRUE/FALSE` isn't entirely stylistic anymore.

**Comment by justinmk** [comment]
`bool` is not stylistic. For one thing, it reduces the surface area of an input to two possible values. That reduces complexity.

**Comment by vheon** [comment]
@justinmk so should I drop the commit and made another PR?

**Comment by justinmk** [comment]
Sure, it'll make it quicker to get this PR done

**Comment by vheon** [comment]
Ok I dropped the `bool` commit and rebased.

**Comment by Pyrohh** [comment]
> bool is not stylistic. For one thing, it reduces the surface area of an input to two possible values. That reduces complexity.

I guess I was wrong:
```c
#include <stdio.h>

int main(void)
{
	_Bool b1 = 1;
	_Bool b2 = 2;
	printf("%d\n%d\n", b1, b2);
	printf("\n%zu\n%zu\n", sizeof(b1), sizeof(b2));
	return 0;
}
```
emits
```
1
1

1
1
```
I forgot about `_Bool` being a built-in type and whatnot, so sorry for the confusion.

File: src/nvim/quickfix.c [hunk 1]
```diff
@@ -1837,10 +1835,12 @@ void qf_age(exarg_T *eap)
     }
   }
 
-  if (eap->addr_count != 0)
-    count = eap->line2;
-  else
+  if (eap->addr_count != 0) {
+    assert(eap->line2 < INT_MAX);
```
Comment (justinmk) [inline src/nvim/quickfix.c]: should be `<=`

File: src/nvim/quickfix.c [hunk 2]
```diff
@@ -2047,7 +2047,7 @@ void ex_copen(exarg_T *eap)
   }
 
   if (eap->addr_count != 0)
-    height = eap->line2;
+    height = (int)eap->line2;
```
Comment (justinmk) [inline src/nvim/quickfix.c]: need an assert here

File: src/nvim/quickfix.c [hunk 3]
```diff
@@ -3355,7 +3356,7 @@ int set_errorlist(win_T *wp, list_T *list, int action, char_u *title)
         vcol,                               /* vis_col */
         pattern,                            /* search pattern */
         nr,
-        type == NULL ? NUL : *type,
+        (char_u)(type == NULL ? NUL : *type),
```
Comment (justinmk) [inline src/nvim/quickfix.c]: This does not match the signature of `qf_add_entry`, this should be `int`.

File: src/nvim/quickfix.c [hunk 4]
```diff
@@ -3319,11 +3320,11 @@ int set_errorlist(win_T *wp, list_T *list, int action, char_u *title)
...
 
     filename = get_dict_string(d, (char_u *)"filename", TRUE);
-    bufnum = get_dict_number(d, (char_u *)"bufnr");
+    bufnum = (int)get_dict_number(d, (char_u *)"bufnr");
     lnum = get_dict_number(d, (char_u *)"lnum");
-    col = get_dict_number(d, (char_u *)"col");
-    vcol = get_dict_number(d, (char_u *)"vcol");
-    nr = get_dict_number(d, (char_u *)"nr");
+    col = (int)get_dict_number(d, (char_u *)"col");
+    vcol = (char_u)get_dict_number(d, (char_u *)"vcol");
```
Comment (justinmk) [inline src/nvim/quickfix.c]: This is passed to a `int` in `qf_add_entry`, not char_u.

**Comment by justinmk** [comment]
@vheon Could you rebase this and check the comments I made?

**Comment by vheon** [comment]
@justinmk Sorry, but since I made this PR I joined a new company, and I should ask permission to collaborate to OSS projects. So at this point I think you're better off closing this PR. I'm really sorry guys. (The same is true for the other PR I made)

**Comment by justinmk** [comment]
@vheon No problem, thanks for followup.

**Comment by pvinis** [comment]
ok i have no idea how to take over and push to this pr. should i make a new pr? or can someone point me to the right direction? i already have a commit in a branch of mine, and i include some changes that are here. thanks.

**Comment by fwalch** [comment]
@pvinis Yes, you need to create a new PR.

**Comment by pvinis** [comment]
alright. thanks again

On Τρί, Σεπ 29, 2015 at 10:34, neovim/neovim
<reply@reply.github.com>
wrote:
@pvinis [https://github.com/pvinis] Yes, you need to create a new PR.

—
Reply to this email directly or view it on GitHub
[https://github.com/neovim/neovim/pull/2818#issuecomment-143987256] .[https://github.com/notifications/beacon/AAGHiRaP6xdUCeYQpePXcTdNOitlkLeZks5o2kSCgaJpZM4E-aGG.gif]

**Pull request closed without merging**